Sources/Breadcrumb.swift

```diff
@@ -8,81 +8,66 @@
-	}
+	/// A class used to represent the breadcrumbs leading up to an events
+	@objc public class Breadcrumb: NSObject {
 
-}
+		// MARK: - Attributes
 
-extension Dictionary {
-	
-	// Sets the key and value but only if value is non-nil
-	func set(key: Key, value: Value?) -> Dictionary<Key, Value> {
-		guard let value = value else { return self }
-		
-		var newDict = self
-		newDict[key] = value
-		
-		return newDict
-	}
-	
-}
+		public let timestamp: NSDate
+		public let category: String
 
-extension Breadcrumb {
-	
-	public convenience init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, level: SentrySeverity? = nil, data: [String: AnyObject]? = nil, to: String, from: String? = nil) {
-		
-		let navigationData = (data ?? [:])
-			.set("to", value: to)
-			.set("from", value: from)
-		
-		self.init(category: category, timestamp: timestamp, message: message, type: "navigation", level: level, data: navigationData)
-	}
-	
-	public convenience init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, level: SentrySeverity? = nil, data: [String: AnyObject]? = nil, url: String, method: String, statusCode: Int? = nil, reason: String? = nil) {
-		
-		let httpData = (data ?? [:])
-			.set("url", value: url)
-			.set("method", value: method)
-			.set("status_code", value: statusCode)
-			.set("reason", value: "reason")
-		
-		self.init(category: category, timestamp: timestamp, message: message, type: "http", level: level, data: httpData)
+		public var type: String?
+		public var message: String?
+		public let data: [String: AnyObject]
+		public let level: SentrySeverity // can't be optional because @objc can't handle optional enums
+
+
+		/// Creates a breadcrumb
+        @objc public init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, type: String? = nil, level: SentrySeverity = .Info, data: [String: AnyObject]? = nil) {
+			self.category = category
+			self.timestamp = timestamp
+			self.message = message
+			self.type = type
+			self.data = data ?? [:]
+			self.level = level
+
+			super.init()
+		}
+
+		/// Conveneince init for a "navigation" type breadcrumb
+        public convenience init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, level: SentrySeverity = .Info, data: [String: AnyObject]? = nil, to: String, from: String? = nil) {
+			let navigationData: [String: AnyObject] = (data ?? [:])
+				.set("to", value: to)
+				.set("from", value: from)
+
+			self.init(category: category, timestamp: timestamp, message: message, type: "navigation", level: level, data: navigationData)
+		}
+
+		/// Conveneince init for an "http" type breadcrumb (-999 workaround because @objc can't handle optional Int)
+        public convenience init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, level: SentrySeverity = .Info, data: [String: AnyObject]? = nil, url: String, method: String, statusCode: Int = -999, reason: String? = nil) {
```

Comment:
If we want this init to be available in `objc` then we can't have `Int?` along with the already mentioned `SentrySeverity?` as a workaround I've set the default to -999 then later check for it and treat it as `nil`